src/disposable/AsyncDisposer.ts

```diff
@@ -1,5 +1,13 @@
 import { IAsyncDisposable } from './types';
 
+const noop = () => {
+  throw new Error('Already disposed');
+};
+
 export class AsyncDisposer implements IAsyncDisposable {
```

Comment:
Do we reallllly need a class here? :laughing: 
This technically may lead to a potential memory leak.

```ts
const disposer = new AsyncDisposer(() => {});
const { _dispose } = disposer;
disposer.dispose();
_dispose(); // the old impl is still accessible
```

Reply:
or use private fields. :trollface:

Reply:
Note that if I do that, TypeScript will no longer let us call it with `new`. So it's another breaking change... and then I should probably lowercase the names to `disposer` and `asyncDisposer` and rename the files, so another breaking change. 🤷 Why not, lets go for it.